Fix: hide full plugin menu in books for network managers

SteelWagstaff opened the pull request:
Network managers can currently see all plugins on the network when they view the plugins page in individual books, which sometimes leads to confusion. See https://github.com/pressbooks/pressbooks-ecampus-ontario/issues/48 for example.

This change makes it so that they see only the 'approved' plugins that book admins can activate/deactivate.

Before:
![Screenshot from 2024-09-04 14-26-07](https://github.com/user-attachments/assets/1e6f4a50-afc6-4f9e-9f62-4affc9ce4592)

After:
![Screenshot from 2024-09-04 14-25-27](https://github.com/user-attachments/assets/ddcb200d-3794-4ab0-af5c-5cd0a92230d6)

SteelWagstaff committed:
fix: hide full plugin menu in books for network managers

arzola reviewed (changes requested):
I think this is not enough to solve the issue

It will still display the Pressbooks Plugins because the foreach will append the approved plugins + pressbooks-* plugins (it will discard all the others but still will include pressbooks ones)

I think we should apply an extra condition in the plugins foreach

I tested it on my local with a network manager and it filters out all other plugins like classic editor, etc... and show pressbooks-* and allowed ones (maybe I'm testing it wrong)

```php
foreach ( $plugins as $slug => $value ) {
			if ( strpos( $slug, 'pressbooks-' ) !== false || in_array( explode( '/', $slug )[0], $slugs, true ) ) {
				$approved[ $slug ] = $value;
			}
		}
```

SteelWagstaff committed:
update if condition

Hide `pressbooks` plugins from network managers and book admins by default

SteelWagstaff committed:
fix tests

Update test now that `pressbooks` plugins are not automatically included in approved list

arzola reviewed (approved):
LGTM
